src/main/scala/intellij/haskell/repl/HaskellConsoleRunner.scala

```diff
@@ -0,0 +1,68 @@
+package intellij.haskell.repl
+
+import java.io.File
+import java.util
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.console.{ConsoleHistoryController, ConsoleRootType, ProcessBackedConsoleExecuteActionHandler}
+import com.intellij.execution.runners.AbstractConsoleRunnerWithHistory
+import com.intellij.execution.{CantRunException, ExecutionException, ExecutionHelper}
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.roots.{ModuleRootManager, ProjectRootManager}
+import intellij.haskell.HaskellNotificationGroup
+import intellij.haskell.sdk.HaskellSdkType
+
+object HaskellConsoleRunner {
+  val REPLTitle = "λ "
+
+  def run(module: Module, fileName: String): Option[HaskellConsoleProcessHandler] = {
+    ModuleRootManager.getInstance(module).getContentRoots.headOption.flatMap(root => {
+      val srcRoot = root.getPath
+      val path = srcRoot + File.separator + "src"
+      val runner = new HaskellConsoleRunner(module, REPLTitle + fileName, path)
+      try {
+        runner.initAndRun()
+        Some(runner.getProcessHandler.asInstanceOf[HaskellConsoleProcessHandler])
+      } catch {
+        case e: ExecutionException =>
+          ExecutionHelper.showErrors(module.getProject, util.Arrays.asList[Exception](e), REPLTitle, null)
+          None
+      }
+    })
+  }
+
+  @throws[CantRunException]
+  private def createCommandLine(module: Module, workingDir: String) = {
+    val sdk = ProjectRootManager.getInstance(module.getProject).getProjectSdk
+    if (sdk == null || !sdk.getSdkType.isInstanceOf[HaskellSdkType] || sdk.getHomePath == null) {
+      HaskellNotificationGroup.logErrorBalloonEvent(module.getProject, "Please make sure your <b>Stack SDK</b> is configured correctly.")
+      throw new CantRunException("Invalid Stack SDK.")
+    } else {
+      new GeneralCommandLine(sdk.getHomePath)
```

Comment:
@rikvdkleij `StackCommandLine.runCommand` and `CommandLine.runProgram` only return `ProcessOutput` instances, but in REPL we have to hold a `Process` instance, but thanks for pointing out to the `StackCommandLine` class, thus lead me a better way to get `stackPath` instead of tedious checking 😄